crates/fluvio-cli/src/consume/mod.rs

```diff
@@ -438,6 +447,11 @@ impl ConsumeOpt {
                                 &mut maybe_terminal_stdout,
                                 &mut maybe_table_model,
                             );
+
+                            if potential_offset == record.offset {
```

Comment:
I think that we could have issues here if we run:

`fluvio consume topic --end-offset 10` in a topic that has more than 10 records. It will never break in that case. Maybe we should do `<=` instead but then we may need to use something different than -1 as default value

Reply:
I see what you're saying here. What if we set disable_continuous to be true if a positive end-offset is specified? That way, if someone accidentally types 55 instead of 5 in a topic with 10 records, at least the loop will break (although they will still get more records than they originally wanted)

Reply:
The problem that I see with that is that disable_continuous will close the stream even if we are not reach the `end offset` that we want.

I think that we can avoid most of these issues with something like:
```rust
if let Some(potencial_offset) = maybe_potencial_offset {
       if  record.offset >= potencial_offset {
          eprintln!("Ending offset has been reached; exiting consumer stream.");
          break;
       }
}
```

Reply:
```
I think that we could have issues here if we run:
fluvio consume topic --end-offset 10 in a topic that has more than 10 records.
```
I'm a bit confused. Did you mean to say less than 10 records here?

For example, say i send 10 records (indexed as 0-9 inclusive) to a new topic (foo, bar, baz, blah, more blah, foo, bar, baz, blah, more blah)

With the code change you provided above (which I completely agree with and have been using), a user inputting `flvd consume example -B --end-offset 9` will see this in the output:
```
foo
bar
baz
blah
more blah
foo
bar
baz
blah
more blah
Ending offset has been reached; exiting
Consumer stream has closed
```

So far, so good. However, if the user inputs `flvd consume example -B --end-offset 10` (ie an offset that is greater than the highest current offset in the topic), then the stream will not be closed and will be waiting on potential new inputs and would require the user to ctrl-c to end consuming. 

Shouldn't we want to close the stream if we've reached the end of the topic even if we inadvertently specify an end-offset that is too high? This is the behavior I was interpreting based on Sehyo and TJ's original comments on the issue. Sehyo mentioned we don't want to wait on further inputs, in which case we'd want to exit early at a specified end-offset or exit when we reach the end of the topic. 

Note that inputting `flvd consume example -B --end-offset 10 -d` would close the stream without waiting for new inputs. In which case, I think it does make sense to force disable_continuous to be true if end-offset is specified, on the off chance someone adds an extra 0 and specifies 100 as an end-offset when they meant 10. There is still the problem of them getting extra values that they didn't want in certain scenarios (for example if they accidentally input 100, really only wanted 10, but got 50 records as the -d flag read until the end of the records on that topic, which happened to be more than 10 but less than 100).

```
The problem that I see with that is that disable_continuous will close the stream even if we havent reached the end offset that we want.
```
I didn't see that behavior on my local cluster. Am I missing something/doing something wrong?

 `flvd consume example -B --end-offset 4 -d` yielded:
```
foo
bar
baz
blah
more blah
```

Reply:
> I'm a bit confused. Did you mean to say less than 10 records here?

I mean, when you use `fluvio consume` without -B, it will fetch the records from the end of the stream. If we run `flvd consume example --end-offset 9` in that topic that you mentioned, it will never break (assuming that we are not using disable_continuous behavior) for the `potential_offset == record.offset` case because the first record that it will get is the record with offset 10.

> So far, so good. However, if the user inputs flvd consume example -B --end-offset 10 (ie an offset that is greater than the highest current offset in the topic), then the stream will not be closed and will be waiting on potential new inputs and would require the user to ctrl-c to end consuming.

I think that this ise the expected behavior, I think that we shouldn't exit early unless we set the disable_continuous flag. Could you confirm which is the expected behavior for this case @sehz?

> I didn't see that behavior on my local cluster. Am I missing something/doing something wrong?

I meant the case that you described above: it will exit earlier if we use as offset a number greater than the highest current offset in the topic.

> (for example if they accidentally input 100, really only wanted 10, but got 50 records as the -d flag read until the end of the records on that topic)

Something that we can do, is t update  the print_status function to include information about --end-offset so the user could get that information just in case that they accidentally put a large input

Reply:
@morenol Luis thanks for your comments. I made some change to force a read from beginning, even if the user forgot the -B flag when specifying an end-offset by changing some logic in the calculate_offset fn. Let me know if you think that makes sense. I am still checking if -B is specified with a particular offset before defaulting to 0.

I also added some logic to ensure a positive end-offset, but noticed that `flvd consume topic -B  --end-offset "-9"` or `flvd consume topic -B  --end-offset -9` already yields `error: Found argument '-9' which wasn't expected, or isn't valid in this context`. So the code I added may be redundant?

I did update print_status as well to specify the end-offset we expect to consume until.

Reply:
start and end offset should be thought of as independent parameters similar to range(min,max).  Of course, basic validation must be enforced (min <= max) and both must be positive offset. 

So we can have the following scenarios.  LEO (Log End Offset) is the last offset in the topic.
(1) LEO = 4,  start_offset = 0 (-B), end_offset =  2,  fetch 0..2
(2) LEO = 4,  start_offset = 4 (no argument),  end_offset = 4,  fetch 4..4
(3) LEO = 4,  start_offset = 4 (no argument),  end_offset = 6,  wait and fetch 4..6
(4) LEO = 4,  start_offset = 2, end_offset = 10,   fetch 2..4, wait, fetch 4.10

Reply:
Thanks @bohlmannc I added some comments to the changes.